tensorflow_graphics/projects/mesh_rcnn/ops/cubify.py

```diff
@@ -0,0 +1,262 @@
+# Copyright 2020 The TensorFlow Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Implementation of the cubify operation for Mesh R-CNN."""
+
+import tensorflow as tf
+
+from tensorflow_graphics.projects.mesh_rcnn.structures import mesh
+from tensorflow_graphics.util import shape
+
+
+def _ravel_index(index, dims):
+  """
+  Computes a linear index in an array of shape dims.
+  Reverse functionality of tf.unravel_index.
+  Args:
+    index: A Tensor with each row corresponding to indices into an array
+    of dimension dims.
+    dims: The shape of the array to be indexed.
+
+  Note:
+
+    The shorthands used below and in the code are
+      `H`: height of input space
+      `W`: width of input space
+      `D`: depth of input space
+
+    Only supports dims=(H,W,D)
```

Comment:
probably switch to Length -X, Width -Y, Height -Z?

Reply:
I've added more information in the cubify docstring about the coordinate system convention. For this function here, I'll leave it as is because _ravel_index is an array-op with row-major ordering.